sdk/core/azure-core/src/main/java/com/azure/core/util/UrlBuilder.java

```diff
@@ -330,7 +342,16 @@ private UrlBuilder with(String text, UrlTokenizerState startState) {
     public URL toUrl() throws MalformedURLException {
         // Continue using new URL constructor here as URI either cannot accept certain characters in the path or
         // escapes '/', depending on the API used to create the URI.
-        return ImplUtils.createUrl(toString());
+        if ("http".equals(scheme)) {
```

Comment:
Either here or where you define `HTTP`/`HTTPS`, maybe a comment noting that the use of a context only works if the protocol matches? (I spent a few minutes clicking through code wondering "why do we need both." Maybe someone else wonders too.)